src/runtime/virtcontainers/container.go

```diff
@@ -580,11 +511,11 @@ func (c *Container) mountSharedDirMounts(ctx context.Context, sharedDirMounts, i
 				return storages, fmt.Errorf("unable to create watchable path: %s: %v", watchableHostPath, err)
 			}
 
-			watchableGuestMount := filepath.Join(kataGuestSharedDir(), "watchable", filepath.Base(guestDest))
+			watchableGuestMount := filepath.Join(kataGuestSharedDir(), "watchable", filepath.Base(sharedFile.guestPath))
```

Comment:
Does watchableGuestMount fit with these reworks?

While it doesn't carry out any additional sharing between the host/guest, it does have some assumptions with respect to paths used on the host/guest, which is otherwise nicely abstracted (I think) into fs_share.

Reply:
I think it does fit, and the guest paths assumptions are now carried through the `SharedFile` return value, which indicates where in the guest the mount is going to live.
Do you think there's anything wrong with the implementation?